metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/query/ESSearchDAO.java

```diff
@@ -262,7 +269,27 @@ public ScrollResult scroll(@Nonnull List<String> entities, @Nonnull String input
 
     scrollRequestTimer.stop();
     // Step 2: execute the query and extract results, validated against document model as well
-    return executeAndExtract(entitySpecs, searchRequest, postFilters, scrollId, keepAlive, size);
+    return executeAndExtract(entitySpecs, searchRequest, transformedFilters, scrollId, keepAlive, size);
+  }
+
+  /**
+   * Allows filtering on entities which are stored as different indices under the hood by transforming the tag
+   * _entityType to _index and updating the type to the index name.
+   * @param filter The filter to parse and transform if needed
+   * @param indexConvention The index convention used to generate the index name for an entity
+   * @return A filter, with the changes if necessary
+   */
+  static Filter transformFilterForEntities(Filter filter, @Nonnull IndexConvention indexConvention) {
+    if (filter != null && filter.getOr() != null) {
+      return new Filter().setOr(new ConjunctiveCriterionArray(filter.getOr().stream().map(
+          conjunctiveCriterion -> new ConjunctiveCriterion().setAnd(conjunctiveCriterion.getAnd().stream().map(
+              criterion -> criterion.getField().equals("_entityType") ? criterion.setField("_index").setValues(
```

Comment:
I think this would be more readable with indents at each stream function call, maybe worth splitting the lambdas into descriptive named functions since there are multiple happening here.

Reply:
Done! I kinda wrote this in a fugue state and left it once it started working... Easier to read now :)

Reply:
Nice! 😄